src/sentry_plugins/bitbucket/client.py

```diff
@@ -10,33 +13,46 @@
 
 
 class BitbucketClient(object):
-    API_URL = u'https://api.bitbucket.org/1.0'
+    API_URL = u'https://api.bitbucket.org/'
 
-    def __init__(self, auth):
+    def __init__(self, auth=None):
         self.auth = auth
 
-    def request(self, method, path, data=None, params=None):
-        oauth = OAuth1(unicode(settings.BITBUCKET_CONSUMER_KEY),
-                       unicode(settings.BITBUCKET_CONSUMER_SECRET),
-                       self.auth.tokens['oauth_token'], self.auth.tokens['oauth_token_secret'],
-                       signature_type='auth_header')
+    def request(self, method, version, path, data=None, params=None, json=True):
+
+        try:
+            oauth = OAuth1(
+                six.text_type(settings.BITBUCKET_CONSUMER_KEY),
+                six.text_type(settings.BITBUCKET_CONSUMER_SECRET),
+                self.auth.tokens['oauth_token'],
+                self.auth.tokens['oauth_token_secret'],
+                signature_type='auth_header'
+            )
+        except KeyError as e:
+            oauth = None
```

Comment:
when would it be ok that oauth is `None`? if there are some requests we need to make without auth, it'd probably be better to be more explicit about it than this. this seems too broad and like stuff could break without our being able to easily tell why